app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go

```diff
@@ -46,32 +49,115 @@ const (
 	// Prefix used by the namer to generate names.
 	// This is used to identify resources created by this code.
 	mciPrefix = "mci1"
+
+	// instanceGroupsAnnotationKey is the annotation key used by gclb ingress
+	// controller to specify the name and zone of instance groups created
+	// for the ingress.
+	// TODO(nikhiljindal): Refactor the code in kubernetes/ingress-gce to
+	// export it and then reuse it here.
+	instanceGroupsAnnotationKey = "ingress.gcp.kubernetes.io/instance-groups"
 )
 
 type LoadBalancerSyncer struct {
 	lbName string
 	hcs    healthcheck.HealthCheckSyncerInterface
+	bss    backendservice.BackendServiceSyncerInterface
 	client kubeclient.Interface
+	igp    ingressig.InstanceGroups
```

Comment:
why igp? Seems like the 'p' isn't necessary. Or perhaps iGrp. (Actually, you set this to gce.GCECloud, so I don't really understand what this member var is for.)